scripts/linters/pylint_extensions.py

```diff
@@ -1893,6 +1893,42 @@ def visit_compare(self, node):
                 self.add_message('inequality-with-none', node=node)
 
 
+class NonTestFilesFunctionNameChecker(checkers.BaseChecker):
+    """Custom pylint checker prohibiting use of "test_only" in function names
+    of non-test files.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'non-test-files-function-name-checker'
+    priority = -1
+    msgs = {
+        'C0031': (
+            'Please refrain from using test_only in function names '
+            'in case of non-test files.',
+            'non-test-files-function-name-checker',
+            'Try using a different name for the function that does not have'
+            'test_only as its prefix.'
+        )
+    }
+
+    def visit_functiondef(self, node):
+        """Visit every function definition and ensure their name doesn't have
+        test_only as its prefix.
+
+        Args:
+            node: astroid.nodes.FunctionDef. A node for a function or method
+                definition in the AST.
+        """
+
+        modnode = node.root()
+        if not modnode.name.endswith('_test'):
+            function_name = node.name
+            if function_name.startswith('test_only'):
+                self.add_message(
+                    'non-test-files-function-name-checker', node=node)
```

Comment:
Return as quick as possible in the top to avoid nested if blocks (use the first if block as guardblock) 
```suggestion
        if modnode.name.endswith('_test'):
            return
        function_name = node.name
        if function_name.startswith('test_only'):
            self.add_message(
                 'non-test-files-function-name-checker', node=node)
```

Reply:
Done.